src/site/topic12.rst

```diff
@@ -11,6 +11,57 @@ Idea #2
 Idea #3
 =======
 
+
+Expand Capacity
+---------------
+
+.. image:: img/arrayqueue_expand0.png
+   :width: 500 px
+   :align: center
+
+* Suppose we ``enqueue`` one more element
+
+.. image:: img/arrayqueue_expand1.png
+   :width: 500 px
+   :align: center
+
+* Now the queue is full
+    * ``size == queue.length``
+
+* If we ``enqueue`` again, we'll need to call ``expandCapacity``
+* **However** we can't just double the size of the array and copy the contents over like we did before
+
+.. Skip arrayqueue_expand2 since it is just a copy of 1
+.. image:: img/arrayqueue_expand3.png
+   :width: 500 px
+   :align: center
+
+* Instead, we should copy the contents into contiguous indices starting at index ``front``
```

Comment:
_Should_ is a little strong here. It doesn't really matter where the queue gets cloned to, so long as the queue stays in order and the indexes point to the correct locations in it.

Reply:
Fixed in the conflict fix main merge